### ISSUE TITLE
[0.13] [Bug] Addition for empty column results in nil value irrespective of fill value

### ISSUE THREAD

**Issue by mattheworiordan** [influxdata/influxdb#6953]
### Bug report

__System info:__ 
0.13, Ubuntu, run in a Docker container

__Steps to reproduce:__

1. Create the data

```
stats,region=eu-west-1 before=5 1467467110000000000
stats,region=eu-west-1 left=0.5 1467467117941000000
stats,region=eu-west-1 right=0.5 1467467118941000000
stats,region=eu-west-1 left=0.5,right=0.5 1467467119941000000
stats,region=eu-west-1 right=0.5 1467467120941000000
stats,region=us-east-1 before=5 1467467110000000000
stats,region=us-east-1 left=5 1467467117941000000
stats,region=us-east-1 right=5 1467467118941000000
stats,region=us-east-1 left=5,right=5 1467467119941000000
stats,region=us-east-1 right=5 1467467120941000000
```

2. Do an initial query in a time range that spans all of the data above

```
select sum(left) + sum(right) + sum(before) from "stats" where time >= 1467467110000000000 and time <= 1467467120941000000 group by region, time(1m) fill(0)
```

Results:

```
-- stats, region:eu-west-1
time	sum_sum_sum
2016-07-02T13:45:00Z	7.5

-- stats, region:us-east-1
time	sum_sum_sum
2016-07-02T13:45:00Z	30
```

This is expected and no surprises here.

3. No do a query that adds a column that does not exist in the time range 

```
select sum(left) + sum(right) + sum(before) from "stats" where time >= 1467467117941000000 and time <= 1467467120941000000 group by region, time(1m) fill(0)
```

```
-- stats, region:eu-west-1
time	sum_sum_sum
2016-07-02T13:45:00Z	

-- stats, region:us-east-1
time	sum_sum_sum
2016-07-02T13:45:00Z	
```

As you can see above, everything is empty

4. Yet if I do the query without adding the columns together, I get the expected values for each column:

```
select sum(left) as "left", sum(right) as "right", sum(before) as "before" from "stats" where time >= 1467467117941000000 and time <= 1467467120941000000 group by region, time(1m) fill(0)
```

```
-- stats, region:eu-west-1
time	left	right	before
2016-07-02T13:45:00Z	1	1.5	

-- stats, region:us-east-1
time	left	right	before
2016-07-02T13:45:00Z	10	15	
```

__Expected behavior:__ 

Calling `select sum(a) + sum(b) group by time(c) fill(0)` should always return the sum of those columns, even if that column happens to not be in the range.

__Actual behavior:__ 

As above, you can see that a missing column in that period causes the addition to fail completely.

**Comment by mattheworiordan** [comment]
For reference, another issue exists as a feature request at https://github.com/influxdata/influxdb/issues/6967.  I do not believe this is a feature, it's a bug

**Comment by jsternberg** [comment]
While I'm not sure that #6967 is a bug, I think this one is. It's a rather insidious one too that doesn't have an easy solution. I think what's happening is that the intervals for the fill iterator are being decided per aggregate rather than for the whole query. Then a nil value ends up becoming part of one of the aggregates and binary math between a value and a null is a null value.

I'll have to investigate more but that sounds like the most plausible explanation to me. We might have to change the behavior of binary math. I think there might be another issue related to this one.

**Comment by jsternberg** [comment]
Related to #6553.

**Comment by jsternberg** [comment]
Closing this in favor of #7216 and this will probably get implemented as part of #3552.
